### DISCUSSION TITLE
Newly added VARCHAR field not returning default value for existing data.

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#246] [General]
In version 6.1.17
I am storing data in spaces of the stream.
I have added new field(VARCHAR) in the stream by using "MODIFY" stream.
Also, I am using CONFIRM CONVERT_DATA while modifying stream

Now when I fetch the data from old spaces with that new field, it is not returning me default value of VARCHAR for old spaces data.
Exception: java.lang.NullPointerException: Parameter specified as non-null is null

Note: it works fine(returning default value) for the new data set loaded after modifying schema.

Can you please suggest what needs to be done here.

**Comment by alex-karpovich** [comment]
@ankitsoin 
did you add new nullable field?

**Reply by ankitsoin** [reply]
I have not mentioned any thing related to nullable, in my updated schema or in old schema.
(Believing by default it is nullable=true, pls correct me if wrong)
I have just added field with "name" 'label' VARCHAR.
This issue is coming only for Varchar columns, when were loaded previously with old schema.

**Reply by alex-karpovich** [reply]
@ankitsoin could you please share exception with stack trace?

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich , 
Please find the below logs if it helps

```
Parameter specified as non-null is null: method com.db.elysium.adapter.timebase.stream.DBTradeMessage.setTrade_id, parameter <set-?>
java.lang.NullPointerException: Parameter specified as non-null is null: method com.db.elysium.adapter.timebase.stream.DBTradeMessage.setTrade_id, parameter <set-?>
	at com.db.elysium.adapter.timebase.stream.DBTradeMessage.setTrade_id(DBTradeMessage.kt)
	at UnknownBOUND_DECODER1.decode(UnknownBOUND_DECODER1.java:81)
	at UnknownBOUND_DECODER1.decode(UnknownBOUND_DECODER1.java:87)
	at com.epam.deltix.qsrv.hf.tickdb.comm.client.TickCursorClient.setupMessage(TickCursorClient.java:1820)
```

Regards,
Shivani

**Reply by alex-karpovich** [reply]
@ShivaniSK If you add a new field using next DDL (for example)

`"ask_price" 'ask size' FLOAT DECIMAL'

it goes as 'NULLABLE' - so your previous data will have **nulls** for this field.

And according to your exception, in your code you expect this field to be non-nullable
```
at com.db.elysium.adapter.timebase.stream.DBTradeMessage.setTrade_id(DBTradeMessage.kt)
```
so you need to change your code or change field to non-nullable

**Comment by ShivaniSK** [comment]
Hello @alex-karpovich,
We are not facing issue with FLOAT DECIMAL types, for FLOAT DECIMAL the default values for previous data is coming as 0.0 when reading from timebase.
We are facing issues with VARCHAR type, where for previous data its showing as NULL.
How do we set a VARCHAR field as non-nullable? Do we need to edit the timebase schema and do it?
